src/hooks/useNetworkMotionStatesAllColonies.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  getExtensionHash,
+  type MotionState,
+  VotingReputationFactory,
+} from '@colony/colony-js';
+import { Extension } from '@colony/colony-js';
+import { useEffect, useMemo, useState } from 'react';
+
+import { ADDRESS_ZERO, supportedExtensionsConfig } from '~constants/index.ts';
+import { useAppContext } from '~context/AppContext/AppContext.ts';
+import { useGetJoinedColoniesExtensionsQuery } from '~gql';
+import { notNull, notUndefined } from '~utils/arrays/index.ts';
+
+export type MotionStatesMap = Map<string, MotionState | null>;
+
+export type MotionStatesMapByColonies = Record<string, MotionStatesMap>;
+
+export type RefetchMotionStates = (motionIdsToRefetch?: string[]) => void;
+
+export type MotionId = {
+  motionId: string;
+  colonyAddress: string;
+  databaseMotionId: string;
+};
+
+const getVotingReputationAddressByColony = (colony) => {
+  const colonyExtensions = colony?.extensions?.items?.filter(notNull);
+  if (!colonyExtensions) {
+    return [];
+  }
+
+  const currentExtensionAddress = colonyExtensions.find((extension) => {
+    const extensionConfig = supportedExtensionsConfig.find(
+      (e) => getExtensionHash(e?.extensionId) === extension?.hash,
+    );
+    return extensionConfig?.extensionId === Extension.VotingReputation;
+  });
+  return currentExtensionAddress?.address;
+};
+
+const useJoinedColoniesWithExtensions = (userAddress?: string) => {
+  const { data, loading } = useGetJoinedColoniesExtensionsQuery({
+    variables: {
+      contributorAddress: userAddress ?? ADDRESS_ZERO,
+    },
+    skip: !userAddress,
+    fetchPolicy: 'cache-and-network',
+  });
+
+  const joinedColoniesWithExtensions = useMemo(() => {
+    return (
+      data?.getContributorsByAddress?.items
+        .filter(notNull)
+        .map((contributor) => contributor.colony) ?? []
+    );
+  }, [data?.getContributorsByAddress?.items]);
+
+  return {
+    joinedColoniesWithExtensions,
+    loading,
+  };
+};
+
+/**
+ * Hook that accepts an array of motion IDs object and returns a map of motion IDs to their states
+ * Make sure to memoize the array of motion IDs to avoid infinite loops
+ */
+const useNetworkMotionStatesAllColonies = (
+  motionIdsMap: MotionId[],
```

Comment:
Instead of a custom type, would passing entire `ColonyMotion` object from `userStake.action.motionData` make things simpler?